include/rcpputils/asserts.hpp

```diff
@@ -57,25 +57,27 @@ class RCPPUTILS_PUBLIC IllegalStateException : public std::exception
  * Checks that an argument condition passes.
  *
  * \param condition
+ * \param msg
  * \throw std::invalid_argument if the condition is not met.
  */
-inline void require_true(bool condition)
+inline void require_true(bool condition, const std::string & msg = "Invalid argument passed!")
```

Comment:
I didn't bring this up on the original pull request, but I keep seeing it, so I'll mention it now. We should probably avoid using `!` in error messages. It's an exception or assertion failure, we already know its important, so I don't think it needs extra punctuation.

I'd recommend following the examples like the STL or glibc, which both use lowercase and no punctuation everywhere. This allows others to combine your error message into a larger string more easily without it looking strange.

For example, consider:

```c++
require_true(false);
```

Depending on how it's handled it might look like this:

```
unhandled exception: invalid argument passed
```

Or:

```
[WARN]: Failed to do X due to invalid argument passed, retrying...
```

If you have punctuation and/or capitalization, it ends up:

```
[WARN]: Failed to do X due to Invalid argument passed!, retrying...
```

It's a really minor thing, but I think it ends up looking more professional to avoid things like exclamation marks, at least in the default messages. If you choose to use them at the user level, that may be totally fine.